notes/advanced-data-types.md

```diff
@@ -0,0 +1,530 @@
+
+## Structures (`structs`)
+
+A `struct` is an entities capable of holding fields, and you can think of it as a way to hold many variables of different types. 
+
+
+Below, we create a struct with the name `myBorland`, with fields `myField1` and `myField2`
+
+```
+/*
+struct {
+    type1 fieldName1;
+    type2 fieldName2;
+    ...
+    typeN fieldNameN;
+} structVarName;
+*/
+
+struct {
+    int myField1;
+    double myField2;
+} myBorland;
+```
+<div style="text-align: center"> Figure 1a - Simple Struct Declaration </div>
+Note how structures are statements, as the ending semicolon denotes. Notice how this structure does not have values yet (Well they do, they're random values, as you'll learn in the Memory and Management Section). So, to initialize values for this `struct`, we will be using dot notation.
+
+```
+// structVarName.fieldName = value;
+
+myBorland.myField1 = 0;
+myBorland.myField2 = 1.0;
+```
+<div style="text-align: center"> Figure 1b - Simple Struct Field Initialization </div>
+
+If you wanted to do this all in one step, then you must specify all values of the enum, seperated with commas, within brackets as shown in the following example:
+```
+struct {
+    int myField1;
+    double myField2;
+} myBorland = {0, 1.0};
+```
+<div style="text-align: center"> Figure 1c - Struct Declaration and Initialization </div>
+
+
+Let's say you wanted to make more structs with the same layout. What you can do is that you can change Figure 1 to name multiple structs, seperated by commas:
+
+```
+/*
+struct {
+    type1 fieldName1;
+    type2 fieldName2;
+    ...
+    typeN fieldNameN;
+} structVarName1, structVarName2, ..., structVarNameM;
+*/
+struct {
+    int myField1;
+    double myField2;
+} myBorland1, myBorland2, myBorland3;
+```
+<div style="text-align: center"> Figure 1d - Multiple Struct Declaration </div>
+
+and the initialization of the fields work exactly the same as in Figure 2!
+
+However, there is a problem with this format. This form of struct formation is really only good for making a single struct. Why? Because if I need this multiple times, chances are I will need this kind of struct again in a different spot in my location, and its not a good idea to reserve memory for a struct that will not get used for a while. So, its good practice to actually declare a type of struct. In the following figure, we've declared the `struct Borland` with the same fields as a new type:
+
+```
+/*
+struct structName {
+    type1 fieldName1;
+    type2 fieldName2;
+    ...
+    typeN fieldnameN;
+};
+/*
+struct Borland {
+    int myField1;
+    double myField2;
+};
+```
+<div style="text-align: center"> Figure 2a - Struct Type Declaration </div>
+
+Note how there is still a (`;`) after the (`}`). Therefore, to make a `Borland`, you can use a declaration and initialization similar to Figures 1 and 2:
+
+```
+// structName structVarName;
+// structName structVarName = {value1, value2, ..., valueN};
+
+struct Borland myBorland1;
```

Comment:
I think it's worth noting that to declare the struct, you need to specify it as a `struct tagName` variable

I.e. you can't just write `Borland myBorland;`

Reply:
I thought so too, but not in C++. See https://learn.microsoft.com/en-us/cpp/cpp/struct-cpp?view=msvc-170. Its good practice though, so I can make a note of that